kratos/utilities/parallel_helpers.cpp

```diff
@@ -0,0 +1,70 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+
+// System includes
+#include <cstdlib>
+#include <thread>
+
+// External includes
+#ifdef KRATOS_SMP_OPENMP
+#include <omp.h>
+#endif
+
+// Project includes
+#include "input_output/logger.h"
+#include "utilities/parallel_helpers.h"
+
+
+namespace Kratos {
+
+int ParallelHelpers::GetNumThreads()
+{
+#if defined(KRATOS_SMP_OPENMP)
+    return omp_get_max_threads();
+
+#elif defined(KRATOS_SMP_CXX11)
+    const char* env_kratos = std::getenv("KRATOS_NUM_THREADS");
```

Comment:
i would get this in the kernel initialization and write it to a kernel variable, so that we don't need to make a system call to get it...(i am not sure of how fast such calls are...)

Reply:
ok